### ISSUE TITLE
move, delete by word

### ISSUE THREAD

**Issue by simonmichael** [jtdaugherty/text-zipper#4]
It would be great to have deleteWord, moveLeftWord, etc. so you could delete and move by word in a brick editor.

**Comment by jtdaugherty** [comment]
Those would indeed be useful. I hesitate to add them to the library because the meaning of "word" will not always be well-understood.  (E.g. in vi, it all depends on what you mean by word, and some keybindings will navigate word boundaries differently.)

With that said, perhaps there are some building blocks missing from the library that would make an implementation of `deleteWord` etc. more straightforward.  I'm thinking of something like

```
deleteWhile :: (Char -> Bool) -> TextZipper a -> TextZipper a
```

and then you could have

```
deleteWord = deleteWhile (not . isSpace)
```

(That wouldn't be smart enough to deal with line endings and other special cases, but you get the idea.)

**Comment by simonmichael** [comment]
Yes, perhaps even a basic implementation, documented as such, would be quite valuable. Brick editor feels very comfortably like editing a command line, until I want to delete a word - then I tend to try M-BS, M-b, C-b, and finally remember LEFT... C-k.

**Comment by jtdaugherty** [comment]
Word movement and editing functions are now available in `0.10`:

The test suite and implementation in `hledger-iadd` are now integrated into `text-zipper` and released:

http://hackage.haskell.org/package/text-zipper-0.10
